src/elements/core/mixins/form-associated-radio-button-mixin.ts

```diff
@@ -0,0 +1,314 @@
+import type { LitElement, PropertyValues } from 'lit';
+import { property } from 'lit/decorators.js';
+
+import { getNextElementIndex, isArrowKeyPressed } from '../a11y.js';
+import { SbbConnectedAbortController } from '../controllers.js';
+import { forceType } from '../decorators.js';
+
+import type { Constructor } from './constructor.js';
+import { SbbDisabledMixin, type SbbDisabledMixinType } from './disabled-mixin.js';
+import {
+  type FormRestoreReason,
+  type FormRestoreState,
+  SbbFormAssociatedMixin,
+  type SbbFormAssociatedMixinType,
+} from './form-associated-mixin.js';
+import { SbbRequiredMixin, type SbbRequiredMixinType } from './required-mixin.js';
+
+export declare class SbbFormAssociatedRadioButtonMixinType
+  extends SbbFormAssociatedMixinType
+  implements Partial<SbbDisabledMixinType>, Partial<SbbRequiredMixinType>
+{
+  public checked: boolean;
+  public disabled: boolean;
+  public required: boolean;
+
+  protected abort: SbbConnectedAbortController;
+
+  public formResetCallback(): void;
+  public formStateRestoreCallback(state: FormRestoreState | null, reason: FormRestoreReason): void;
+
+  protected isDisabledExternally(): boolean;
+  protected isRequiredExternally(): boolean;
+  protected withUserInteraction?(): void;
+  protected updateFormValue(): void;
+  protected updateFocusableRadios(): void;
+  protected emitChangeEvents(): void;
+  protected navigateByKeyboard(radio: SbbFormAssociatedRadioButtonMixinType): Promise<void>;
+}
+
+/**
+ * A static registry that holds a collection of `radio-buttons`, grouped by `name`.
+ * It is mainly used to support the standalone groups of radios.
+ * @internal
+ */
+export class RadioButtonRegistry {
```

Comment:
I thought about it but still chose a "Singleton / Static" service approach because I feel that's closer to the concept of a "global registry." 
Apart from connecting/disconnecting, the Controller would not be able to take care of much anyway.
I'm not deadset on this decision, though. I'm open to discussion

Reply:
I think the inertController was very similar, as it also has a global state but still is a Reactive Controller. But yeah the base is the same ;-) Let's open the discussion to @kyubisation

Reply:
This is mega nitpicky, but I think this can be solved with WeakMaps and a Set.

e.g. 
```ts
// on module level
const radioButtonRegistry = new WeakMap<Node, WeakMap<string, Set<SbbFormAssociatedRadioButtonMixinType>>>

...

private _associatedRadioButtons: Set<SbbFormAssociatedRadioButtonMixinType>;

...

private _connectToRegistry() {
  // The following code should probably only be executed client side
  const associationElement = this.closest('form') ?? this.getRootNode();
  let nameMap = radioButtonRegistry.get(associationElement);
  if (!nameMap) {
    nameMap = new WeakMap();
    radioButtonRegistry.set(associationElement, nameMap);
  }
  this._associatedRadioButtons = nameMap.get(this.name);
  if (!this._associatedRadioButtons) {
    this._associatedRadioButtons = new Set();
    nameMap.set(this.name, this._associatedRadioButtons);
  } 
  this._associatedRadioButtons.add(this);
}

private _disconnectFromRegistry() {
  this._associatedRadioButtons?.delete(this);
}
```

@TomMenga What do you think?

Reply:
The only real difference from the proposal is that the name registry has to be a `Map` (since `WeakMap` only allows non-primitive types as keys).
In theory, this will prevent the second level from being garbage collected until the `Node` key is also marked (which might never be the case if the Node is the `document`)

Alternatively, in the `_disconnectFromRegistry` we could manually delete the `<string, Set>` entry if the Set is empty. Not sure if it's worth though.

@kyubisation @jeripeierSBB What do you think?

```js
const radioButtonRegistry = new WeakMap<Node, Map<string, Set<SbbFormAssociatedRadioButtonMixinType>>>
```

Reply:
Ah, I did not think about that. Yes, Map should be ok.
I think your suggestion to remove the key when the Set is empty, is probably the cleanest.
Although then we would have to cache the name Map as well, as we cannot rely on `this.closest('form') ?? this.getRootNode()` when being disconnected. But it should be fine to cache both map and set for this scenario.